Examples/Framework/include/ActsExamples/EventData/Measurement.hpp

```diff
@@ -28,253 +29,357 @@
-  /// @param subspaceIndices Which parameters are measured
-  /// @param params Measured parameters values
-  /// @param cov Measured parameters covariance
-  ///
-  /// @note The indices must be ordered and must describe/match the content
-  ///   of parameters and covariance.
-  template <typename other_indices_t, std::size_t kSize, typename parameters_t,
-            typename covariance_t>
-  VariableSizeMeasurement(
-      Acts::SourceLink source,
-      const std::array<other_indices_t, kSize>& subspaceIndices,
-      const Eigen::MatrixBase<parameters_t>& params,
-      const Eigen::MatrixBase<covariance_t>& cov)
-      : m_source(std::move(source)) {
-    static_assert(kSize == parameters_t::RowsAtCompileTime,
-                  "Parameter size mismatch");
-    static_assert(kSize == covariance_t::RowsAtCompileTime,
-                  "Covariance rows mismatch");
-    static_assert(kSize == covariance_t::ColsAtCompileTime,
-                  "Covariance cols mismatch");
-
-    m_subspaceIndices.resize(subspaceIndices.size());
-    std::transform(subspaceIndices.begin(), subspaceIndices.end(),
-                   m_subspaceIndices.begin(), [](auto index) {
-                     return static_cast<SubspaceIndex>(index);
-                   });
-
-    parameters<kSize>() = params;
-    covariance<kSize>() = cov;
+  template <std::size_t Size>
+  using FixedProxy = FixedMeasurementProxy<Acts::eBoundSize, Size, false>;
+  template <std::size_t Size>
+  using ConstFixedProxy = FixedMeasurementProxy<Acts::eBoundSize, Size, true>;
+  using VariableProxy = VariableMeasurementProxy<Acts::eBoundSize, false>;
+  using ConstVariableProxy = VariableMeasurementProxy<Acts::eBoundSize, true>;
+
+  MeasurementContainer();
+
+  std::size_t size() const;
+
+  void reserve(std::size_t size);
+
+  std::size_t addMeasurement(std::uint8_t size);
+
+  VariableProxy getMeasurement(std::size_t index);
+  ConstVariableProxy getMeasurement(std::size_t index) const;
+
+  template <std::size_t Size>
+  FixedProxy<Size> getMeasurement(std::size_t index) {
+    return FixedProxy<Size>{*this, index};
+  }
+  template <std::size_t Size>
+  ConstFixedProxy<Size> getMeasurement(std::size_t index) const {
+    return ConstFixedProxy<Size>{*this, index};
   }
-  /// A measurement can only be constructed with valid parameters.
-  VariableSizeMeasurement() = delete;
-  VariableSizeMeasurement(const VariableSizeMeasurement&) = default;
-  VariableSizeMeasurement(VariableSizeMeasurement&&) = default;
-  ~VariableSizeMeasurement() = default;
-  VariableSizeMeasurement& operator=(const VariableSizeMeasurement&) = default;
-  VariableSizeMeasurement& operator=(VariableSizeMeasurement&&) = default;
 
-  /// Source link that connects to the underlying detector readout.
-  const Acts::SourceLink& sourceLink() const { return m_source; }
+  VariableProxy makeMeasurement(std::uint8_t size);
+  template <std::size_t Size>
+  FixedProxy<Size> makeMeasurement() {
```

Comment:
ah this is the ideomatic way to create measurements... does `addMeasurement` needs to be public actually?